crates/sui-node/src/lib.rs

```diff
@@ -446,17 +446,42 @@ impl SuiNode {
         epoch_store: Arc<AuthorityPerEpochStore>,
         checkpoint_store: Arc<CheckpointStore>,
         state_sync_handle: state_sync::Handle,
+        p2p_network: Network,
         registry_service: &RegistryService,
     ) -> Result<ValidatorComponents> {
+        let system_state = state
```

Comment:
Unfortunately it is not safe to use `get_sui_system_state_object` in this place as it does not guarantee to hold system state for current epoch.

Instead, to expose information you are lookin for, we could do the following:

(1) There is `EpochStartConfigration` - you can add fields that you are interested in to this object.
(2) You can find a code where it is initialized and pass more data from `SuiSystemObject` into `EpochStartConfigration`. This is the only place in code where access to `SuiSystemObject` is safe.
(3) Here in SuiNode:: construct_validator_components you can use `epoch_store.get_epoch_start_configuration` safely to get the data that you need.